src/main/java/org/mule/module/json/internal/JsonSchemaValidator.java

```diff
@@ -7,356 +7,102 @@
 
-    private JsonSchema loadSchema(JsonSchemaFactory factory) {
-      try {
-        checkState(schemaLocation != null, "schemaLocation has not been provided");
-        String realLocation = resolveLocationIfNecessary(formatUri(schemaLocation));
-        return factory.getJsonSchema(realLocation);
-      } catch (Exception e) {
-        throw new ModuleException(format("Could not load JSON schema [%s]. %s", schemaLocation, e.getMessage()),
-                                  SCHEMA_NOT_FOUND, e);
-      }
+    if (!allowDuplicateKeys) {
+      objectMapper.enable(STRICT_DUPLICATE_DETECTION);
     }
 
-    private JsonSchema loadSchema(String schemaContent) {
-      try {
-        JsonNode schemaNode = JsonLoader.fromString(schemaContent);
-        JsonSchemaFactory factory = JsonSchemaFactory.byDefault();
-        return factory.getJsonSchema(schemaNode);
-      } catch (ProcessingException e) {
-        throw new ModuleException("Invalid Schema", INVALID_SCHEMA, e);
-      } catch (IOException e) {
-        throw new ModuleException("Invalid Input Content", INVALID_INPUT_JSON, e);
-      }
+    if (allowArbitraryPrecision) {
+      objectMapper.enable(USE_BIG_DECIMAL_FOR_FLOATS);
     }
 
-    private String resolveLocationIfNecessary(String path) {
-      URI uri = URI.create(path);
-
-      String scheme = uri.getScheme();
-      if (scheme == null || "resource".equals(scheme)) {
-        return openSchema(uri.getPath()).toString();
-      }
-      return path;
-    }
-
-    private URL openSchema(String path) {
-      URL url = Thread.currentThread().getContextClassLoader().getResource(path);
-      if (url == null && path.startsWith("/")) {
-        url = openSchema(path.substring(1));
-        if (url != null) {
-          return url;
-        }
-      }
-
-      if (url == null) {
-        throw new IllegalArgumentException("Cannot find schema [" + path + "]");
-      }
-
-      return url;
+    if (Boolean.parseBoolean(System.getProperty(VALIDATOR_FAIL_ON_TRAILING_TOKENS, "false"))) {
+      objectMapper.enable(FAIL_ON_TRAILING_TOKENS);
     }
 
-    private String formatUri(String location) {
-      URI uri = URI.create(location);
-
-      if (uri.getScheme() == null) {
-        if (location.charAt(0) == '/') {
-          location = location.substring(1);
-        }
-
-        location = RESOURCE_PREFIX + location;
-      }
-
-      return location;
+    for (Map.Entry<String, String> redirect : redirects.entrySet()) {
+      checkArgument(!isBlank(redirect.getKey()), "from cannot be null or blank");
+      checkArgument(!isBlank(redirect.getValue()), "to cannot be null or blank");
+      schemaRedirects.put(formatUri(redirect.getKey()), formatUri(redirect.getValue()));
     }
   }
 
-
-  /**
-   * Returns a new {@link Builder}
-   *
-   * @return a {@link Builder}
-   */
-  public static Builder builder() {
-    return new Builder();
+  public void validate(InputStream inputStream) {
+    // do nothing
```

Comment:
Is this really the default behavior or should this be and abstract method?

Reply:
Should be an abstract method 👍